airflow/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -99,67 +97,29 @@ def __init__(
             )
         self.aws_conn_id = aws_conn_id
         self.region = region
-        self.statement_id = None
+        self.statement_id: str | None = None
 
     @cached_property
     def hook(self) -> RedshiftDataHook:
         """Create and return an RedshiftDataHook."""
         return RedshiftDataHook(aws_conn_id=self.aws_conn_id, region_name=self.region)
 
-    def execute_query(self):
-        kwargs: dict[str, Any] = {
```

Comment:
This also not a good idea to remove any public methods. Instead of remove we need to deprecate it and keep backward compatibility for a while.

Some of users might create own operators based on `RedshiftDataOperator` and use this methods.